### PR TITLE
Fix bug that was preventing the API from picking up cover expiration events

### PR DESCRIPTION
**Pull request by roxdanila**

### CODE REVIEW

**Comment by roxdanila** [comment]
@MilGard91 can you test this out? and eventually add some programatic tests for it

**Comment by MilGard91** [comment]
Since ethers.js does't allow to create one listener to multiple events in one go ([issue](https://github.com/ethers-io/ethers.js/discussions/3144)) we had to divide them it into multiple listeners

**Review by MilGard91 — APPROVED** [review]

File: test/unit/eventApi.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+          acc[key] = value;
+        }
+        return acc;
+      }, {});
+      Object.setPrototypeOf(stakingPoolFactoryMock, Object.getPrototypeOf(stakingPoolFactory));
+      return stakingPoolFactoryMock;
+    }
+    return contracts(name, id, forceNew);
+  };
+}
```
Comment (shark0der) [inline test/unit/eventApi.js:34]: is it possible to use a js proxy instead?

File: test/unit/eventApi.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+describe.only('Catching events', () => {
```
Comment (shark0der) [inline test/unit/eventApi.js:36]: forgotten `.only`?

File: test/unit/eventApi.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    });
+    for (const eventName of stakingProductsEvents) {
+      stakingProducts.emit(eventName);
+    }
+
+    // pushing expect to end of the event queue
+    setTimeout(() => {
+      expect(eventCounter).to.be.equal(stakingProductsEvents.length);
+      expect(productChangeCounter).to.be.equal(stakingProductsEvents.length);
+    }, 0);
```
Comment (shark0der) [inline test/unit/eventApi.js:115]: This will execute after the test passes so it will never catch a failure

File: test/unit/eventApi.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    });
+    for (const eventName of stakingPoolFactoryEvents) {
+      stakingPoolFactory.emit(eventName);
+    }
+
+    // pushing expect to end of the event queue
+    setTimeout(() => {
+      expect(eventCounter).to.be.equal(stakingPoolFactoryEvents.length);
+      expect(poolChangeCounter).to.be.equal(stakingPoolFactoryEvents.length);
+    }, 0);
```
Comment (shark0der) [inline test/unit/eventApi.js:138]: ditto

File: test/unit/eventApi.js (hunk 3 above)
Comment (shark0der) [reply test/unit/eventApi.js:115]: Seems like I'm wrong, but I'd call it a race condition anyway

**Pull request merged**